### ISSUE TITLE
High CPU usage on OSX when hitting Fn+Left (Home) or Fn+Right (end)

### ISSUE THREAD

**Issue by darkguy2008** [Microsoft/vscode#38975]
- VSCode Version: Code - Insiders 1.19.0-insider (3afa3348c5cc8ee7fb29458e303cf476b0887f0a, 2017-11-22T08:33:33.249Z)
- OS Version: Darwin x64 17.2.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
vscode-icontheme-nomo-dark|be5|1.3.4
csharp|ms-|1.13.1
vscode-code-outline|pat|0.0.12
java|red|0.14.0
sass-indented|rob|1.4.5
vscode-java-debug|vsc|0.3.1
vscode-java-pack|vsc|0.2.0
vscode-todo-highlight|way|0.5.11

---

Steps to Reproduce:

1. Use VSCode Insiders on OSX
2. Hit Fn+Left or Fn+Right

Reproduces without extensions: Yes

![vscode1](https://user-images.githubusercontent.com/6304136/33143001-062eacec-cf7e-11e7-8e87-93553bd639bd.gif)

This also happens with Stable version. Could it be my Mac? :( :(

**Comment by rebornix** [comment]
@darkguy2008 thanks for reporting. I tried really hard to reproduce but can only push the CPU usage to around 13%. Since you can reproduce it without any extension running, could you please profile the window by following https://github.com/Microsoft/vscode/wiki/Performance-Issues#profile-a-vs-code-window and send me the results for better troubleshooting?

**Comment by darkguy2008** [comment]
Thanks @rebornix for your answer. I've attached the profiles here. SInce it also happens with Stable I also ran the profiling against it. Hope it's useful :)

[Profiles Stable & Insiders.zip](https://github.com/Microsoft/vscode/files/1497129/Profiles.Stable.Insiders.zip)

**Comment by rebornix** [comment]
@darkguy2008 thanks for your quick response! From the perf log, the rendering process is doing great. Can you try https://github.com/Microsoft/vscode/issues/21580#issuecomment-283317572 and provide me the info like which process (with its arguments) is eating the most CPU?

You can open Activity Monitor, Menu -> View -> All Processes, Hierarchically, get the pid of that bad process and run `ps aux | grep <pid>`. Thanks again for your help!

**Comment by darkguy2008** [comment]
Hello @rebornix, you're welcome!. Indeed it's VSCode, see:

![image](https://user-images.githubusercontent.com/6304136/33152550-c4c87a1a-cfa2-11e7-854c-fe051b281b32.png)

What else can I help with?

**Comment by darkguy2008** [comment]
Could it be related to High Sierra? Also, I found out something: My hackintosh has Sierra and no issues with it. This Macbook got updated to High Sierra about two weeks ago, and I've started to experience this problem a couple of days ago.

I've also found out that if I hit Fn+Left/Right it goes 99% CPU usage, but once the caret has moved, **as long as I repeat the keys Fn+Left or Right in spans of 0.5ms each, it works. If I do it faster, it hangs as shown in the screenshots and GIF image.

Any ideas? :)

**Comment by rebornix** [comment]
@darkguy2008 from what you described, it's likely an Electron/High Sierra issue

**Comment by darkguy2008** [comment]
@rebornix I see, thanks for your confirmation. Is there any way I can be of help? Most of us in OSX are migrating to High Sierra and this will be an important issue. I have some Electron experience but I don't know how to create a small project to replicate it and send it to the Electron developers. Any ideas? :)

**Comment by darkguy2008** [comment]
Alright, while it's been an ordeal to get any development suite going, I went with Brackets (easier to setup than Atom, didn't try VSCode yet as the issue happens in Brackets as well) and I found out that the delay goes further up the chain from DOM events as well. The issue could be lingering between Electron itself and the OS. Here's what I get when I hit the same key combo, setting an event breakpoint on any key-related event:

![brackets1](https://user-images.githubusercontent.com/6304136/33236873-dfc74478-d226-11e7-89cb-e50a05e164d7.gif)

**Comment by darkguy2008** [comment]
I managed to reproduce it with a simple Electron app. So it's Electron, unrelated to VSCode but still an issue in High Sierra. Will try to report there and reference this issue then ;)

**Comment by darkguy2008** [comment]
Hi all, just wanted to mention that the problem seemed to be the Onyx app I ran a few days ago before I reported the bug ( https://www.titanium-software.fr/en/onyx.html ) DON'T RUN IT IN HIGH SIERRA. It will screw up your install bad time.

I fixed it by running CleanMyMac 3 and running all the maintenance scripts (ticked all checkboxes). Rebooted, and issue got fixed.

This can be closed :D (still, it's an interesting issue to check though).
